src/framework/pybind_json.hpp

```diff
@@ -271,4 +331,230 @@ void std::from_json(const json_t &js, py::object &o) {
+  }
+  for (const auto &pair : result.additional_cvector_data_) {
+    pyresult[pair.first.data()] = pair.second;
+  }
+  for (const auto &pair : result.additional_cmatrix_data_) {
+    pyresult[pair.first.data()] = AERPy::from_matrix(pair.second);    
+  }
+
+  // Snapshot data
+  if (result.return_snapshots_) {
+    py::dict snapshots;
+    // Average snapshots
+    for (const auto &pair : result.average_json_snapshots_) {
+      py::object tmp;
+      from_json(pair.second, tmp);
+      snapshots[pair.first.data()] = tmp;
+    }
+    for (auto &pair : result.average_complex_snapshots_) {
+      snapshots[pair.first.data()] = AERPy::from_avg_snap(pair.second);
+    }
+    for (auto &pair : result.average_cvector_snapshots_) {
+      snapshots[pair.first.data()] = AERPy::from_avg_snap(pair.second);
+    }
+    for (auto &pair : result.average_cmatrix_snapshots_) {
+      snapshots[pair.first.data()] = AERPy::from_avg_snap(pair.second);
+    }
+    for (auto &pair : result.average_cmap_snapshots_) {
+      snapshots[pair.first.data()] = AERPy::from_avg_snap(pair.second);
+    }
+    for (auto &pair : result.average_rmap_snapshots_) {
+      snapshots[pair.first.data()] = AERPy::from_avg_snap(pair.second);
+    }
+    // Singleshot snapshot data
+    // Note these will override the average snapshots
+    // if they share the same type string
+    for (const auto &pair : result.pershot_json_snapshots_) {
+      py::object tmp;
+      from_json(pair.second, tmp);
+      snapshots[pair.first.data()] = tmp;
+    }
+    for (auto &pair : result.pershot_complex_snapshots_) {
+      py::dict d;
+      // string PershotData
+      for (auto &per_pair : pair.second.data())
+        d[per_pair.first.data()] = per_pair.second.data();
+      snapshots[pair.first.data()] = d;
+    }
+    for (auto &pair : result.pershot_cvector_snapshots_) {
+      py::dict d;
+      // string PershotData
+      for (auto &per_pair : pair.second.data())
+        d[per_pair.first.data()] = per_pair.second.data();
+      snapshots[pair.first.data()] = d;
+    }
+    for (auto &pair : result.pershot_cmatrix_snapshots_) {
+      py::dict d;
+      // string PershotData
+      for (auto &per_pair : pair.second.data())
+        d[per_pair.first.data()] = per_pair.second.data();
+      snapshots[pair.first.data()] = d;
+    }
+    for (auto &pair : result.pershot_cmap_snapshots_) {
+      py::dict d;
+      // string PershotData
+      for (auto &per_pair : pair.second.data())
+        d[per_pair.first.data()] = per_pair.second.data();
+      snapshots[pair.first.data()] = d;
+    }
+    for (auto &pair : result.pershot_rmap_snapshots_) {
+      py::dict d;
+      // string PershotData
+      for (auto &per_pair : pair.second.data())
+        d[per_pair.first.data()] = per_pair.second.data();
+      snapshots[pair.first.data()] = d;
+    }
+    if ( py::len(snapshots) != 0 )
+        pyresult["snapshots"] = snapshots;
+  }
+  //for (auto item : pyresult)
```

Comment:
nit:  Remove these commented lines

Reply:
I think this is kind of useful for any debugging - the pybind print function is useful (especially here) but has a weird syntax and I often forgot it was there